### ISSUE TITLE
Handle wildcards when trying to match the best server name

### ISSUE THREAD

**Issue by ehsan** [letsencrypt/letsencrypt#2240]
Currently if you have an apache vhost file with a `VirtualHost` containing:

```
  ServerName site.com
  ServerAlias *.site.com
```

And pass `-d site.com,www.site.com` to the Let's Encrypt client, you get a "We were unable to find a vhost with a ServerName or Address of www.site.com" error message and even worse get a selection table.

This is because [_find_best_vhost()] (https://github.com/letsencrypt/letsencrypt/blob/master/letsencrypt-apache/letsencrypt_apache/configurator.py#L341) is completely unaware of aliases.

**Comment by patf** [comment]
[get_names()](https://github.com/letsencrypt/letsencrypt/blob/6fbcebd4ab8cb356de92c80c976c9bf0fefb13b4/letsencrypt-apache/letsencrypt_apache/obj.py#L130) seems to handle aliases. I think the issue here is that the client doesn't match wildcards in any way.

I suppose the right question to ask is whether it makes sense to support this given the fact that Let's Encrypt itself doesn't support wildcard certificates. Users would have to enumerate all domains when they call the client anyway, so expecting the same from your apache configuration is fine IMO. This would also further complicate the matching logic, as wildcard aliases could intersect with regular names and thus require additional weighing logic.

**Comment by ehsan** [comment]
> get_names() seems to handle aliases. I think the issue here is that the client doesn't match wildcards in any way.

You're right!

> I suppose the right question to ask is whether it makes sense to support this given the fact that Let's Encrypt itself doesn't support wildcard certificates. Users would have to enumerate all domains when they call the client anyway, so expecting the same from your apache configuration is fine IMO. This would also further complicate the matching logic, as wildcard aliases could intersect with regular names and thus require additional weighing logic.

Well, this will force users to have bloated config files, like:

```
  ServerAlias *.site.com
  ServerAlias every.site.com
  ServerAlias single.site.com
  ServerAlias prefix.site.com
```

It's also surprising since if you know how ServerAlias works in Apache, it's not that obvious why the client complains about some but not all of your domain names.  It took me ~half an hour to figure out what was wrong and that was only after I ended up looking at the code.  :-)

**Comment by pde** [comment]
I think it's probably correct to match the wildcards when deciding if a vhost is the one we want to work with, but we might want to add those ServerAliases ourselves as we install certs (along the same lines as #1991).

**Comment by SwartzCr** [comment]
so my change that added `_add_servername_alias` will always add explicit `ServerAlias` lines to the generated conf file - so all we need to do is check for *s in `_find_best_vhost`

**Comment by bmw** [comment]
fixed in #2537
